### PR TITLE
Refactoring/test runtime creation

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Removes the `TestUtil` helper and replaces it's usage with `Resources.mkRuntime` where possible. This will make plugging in the new RSpace much more pleasant

### JIRA ticket:
Prerequisite for: https://rchain.atlassian.net/browse/RCHAIN-3327

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by lukasz-golebiewski** [comment]
bors try

File: rholang/src/test/scala/coop/rchain/rholang/Resources.scala [hunk 1]
```diff
@@ -60,27 +62,22 @@ object Resources {
...
-      storageSize: Long = 1024 * 1024,
-      storeType: StoreType = StoreType.LMDB
-  )(
-      implicit log: Log[Task],
-      scheduler: Scheduler,
-      metrics: Metrics[Task]
-  ): Resource[Task, Runtime[Task]] =
-    mkTempDir[Task](prefix)
+      storageSize: Long = 1024 * 1024
+  )(implicit scheduler: Scheduler): Resource[F, Runtime[F]] =
```
Comment (rabbitonweb) [inline rholang/src/test/scala/coop/rchain/rholang/Resources.scala]: I wonder why scheduler is needed. Compiler is not complaining so I guess it's there for a reason.

File: casper/src/test/scala/coop/rchain/casper/helper/RhoSpec.scala [hunk 2]
```diff
@@ -61,23 +65,30 @@ object RhoSpec {
...
-                )
-
-      rand = Blake2b512Random(128)
-      _ <- TestUtil
-            .eval(testObject.code, runtime)(implicitly, rand.splitShort(1))
-            .timeout(timeout)
-
-      result <- testResultCollector.getResult
-    } yield result
+    TestResultCollector[Task].flatMap { testResultCollector =>
```
Comment (sebtomba) [inline casper/src/test/scala/coop/rchain/casper/helper/RhoSpec.scala]: This `flatMap` looks pretty strange

**Comment by lukasz-golebiewski** [comment]
bors retry

File: casper/src/test/scala/coop/rchain/casper/helper/RhoSpec.scala (hunk 2 above)
Comment (lukasz-golebiewski) [reply casper/src/test/scala/coop/rchain/casper/helper/RhoSpec.scala]: Why is that?

**Pull request merged**